components/SinglePageComponents/Comments.js

```diff
@@ -1,16 +1,43 @@
 import { useState, useEffect } from "react";
 import styled from "styled-components";
+import Image from "next/image";
+import STAR_FILLED from "../../public/images/star_filled.png";
 
-export default function Comments({ id }) {
+export default function Comments({ currentRouteID }) {
   const [name, setName] = useState("");
   const [comment, setComment] = useState("");
   const [comments, setComments] = useState([]);
   const [rating, setRating] = useState(0);
 
+  async function performFetch() {
+    const allComments = await getComments();
+    setComments(allComments);
+  }
+
   useEffect(() => {
-    const comments = JSON.parse(localStorage.getItem(`comments-${id}`)) || [];
-    setComments(comments);
-  }, [id]);
+    performFetch();
+  }, []);
+
+  //----GET
+
+  async function getComments() {
+    try {
+      const response = await fetch(`/api/comments`);
+      const data = await response.json();
+      const routeComments = data.filter((comment) => {
+        if (comment.routeID === currentRouteID) {
+          return comment;
+        } else {
+          return null;
+        }
+      });
```

Comment:
```suggestion
      const routeComments = data.filter((comment) => {
          return comment.routeID === currentRouteID
        }
      });
```

ich glaub das geht auch so